.github/test-reporter.yml

```
@@ -0,0 +1,20 @@
+---
+## Workflow to process the JUnit test results and add a report to the checks.
+name: Test Report
+on:
+  workflow_run:
+    workflows:
+      - test
+    types:
+      - completed
+
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/test-report@gh_actions
+        with:
+          artifact: test-results            # artifact name
+          name: JUnit Tests                 # Name of the check run which will be created
```

```suggestion
          name: Test Results                 # Name of the check run which will be created
```

IIUC, it would otherwise say "JUnit Tests" in the GitHub checks UI. That would be quite confusing, since the tests are written in Go - we're just converting them to JUnit format for the report formatter.